pkg/jira/customfield.go

```diff
@@ -1,41 +1,88 @@
 package jira
 
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
+
 const (
+	customFieldFormatAny     = "any"
 	customFieldFormatOption  = "option"
 	customFieldFormatArray   = "array"
 	customFieldFormatNumber  = "number"
+	customFieldFormatString  = "string"
 	customFieldFormatProject = "project"
+	customFieldFormatJson    = "json"
 )
 
 type customField map[string]interface{}
 
 type customFieldTypeNumber float64
 
-type customFieldTypeNumberSet struct {
-	Set customFieldTypeNumber `json:"set"`
+type customFieldTypeString string
+
+type customFieldTypeOption struct {
+	Value string `json:"value"`
 }
 
-type customFieldTypeStringSet struct {
-	Set string `json:"set"`
+type customFieldTypeProject struct {
+	Value string `json:"key"`
 }
 
-type customFieldTypeOption struct {
-	Value string `json:"value"`
+type customFieldTypeJson struct {
+	Json string
 }
 
-type customFieldTypeOptionSet struct {
-	Set customFieldTypeOption `json:"set"`
+func (field customFieldTypeJson) MarshalJSON() ([]byte, error) {
+	return []byte(field.Json), nil
 }
 
-type customFieldTypeOptionAddRemove struct {
-	Add    *customFieldTypeOption `json:"add,omitempty"`
-	Remove *customFieldTypeOption `json:"remove,omitempty"`
+type customFieldEditTypeSet struct {
+	Set any `json:"set"`
 }
 
-type customFieldTypeProject struct {
-	Value string `json:"key"`
+type customFieldEditTypeAddRemove struct {
+	Add    *any `json:"add,omitempty"`
+	Remove *any `json:"remove,omitempty"`
 }
 
-type customFieldTypeProjectSet struct {
-	Set customFieldTypeProject `json:"set"`
+func constructCustomField(dataType string, itemType string, value string) any {
+	switch dataType {
+	case customFieldFormatOption:
+		return customFieldTypeOption{Value: value}
+	case customFieldFormatProject:
+		return customFieldTypeProject{Value: value}
+	case customFieldFormatArray:
+		pieces := strings.Split(strings.TrimSpace(value), ",")
+		items := make([]any, len(pieces))
+		for idx, piece := range pieces {
+			items[idx] = constructCustomField(itemType, piece, "")
```

Comment:
```suggestion
			items[idx] = constructCustomField(itemType, "", piece)
```


We seem to be setting empty value here. That breaks the following:

```sh
$ jira issue create -s"Some story" --custom platform=Linux --debug
```